web_session_allow_public/tests/test_ir_http.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+import mock
+
+from openerp.tests.common import HttpCase
+
+from openerp.addons.base.ir.ir_http import ir_http
+
+
+MODULE_PATH = 'openerp.addons.base.ir.ir_http'
+
+
+class TestIrHttp(HttpCase):
+
+    @mock.patch('%s.request' % MODULE_PATH)
+    def test_public_session(self, request):
+        """ It should allow HTTP authentication on public user """
+        request.session.uid = False
+        with mock.patch.object(
```

Comment:
You are the `mock` guy indeed :laughing: 

But, did you know you can actually make RPC calls in tests? [you have some helpers inside `HttpCase`](https://github.com/odoo/odoo/blob/9.0/openerp/tests/common.py#L246-L249) and [an example here](https://github.com/odoo/odoo/blob/fc2e80cb4bcc450762c7ac5cb82a3e2d88062b38/odoo/addons/base/tests/test_xmlrpc.py).